pkgs/applications/science/math/mathematica/default.nix

```diff
@@ -93,6 +94,12 @@ stdenv.mkDerivation rec {
 
     echo "=== Running MathInstaller ==="
     ./MathInstaller -auto -createdir=y -execdir=$out/bin -targetdir=$out/libexec/Mathematica -silent
+
+    # Fix library paths
```

Comment:
```fish
% pwd
/nix/store/zldjril1j51qy4zjz9442cgrhh3wvg4v-mathematica-11.1.1-ja/libexec/Mathematica/SystemFiles
% for dir in (find . -type d) 
      cd $dir
      for file in (find . -type f -perm -0100)
          ldd $file | rg "`ZLIB_1.2.9' not found"
      end
      prevd
  end
cd: The directory “./Formats” does not exist
cd: The directory “./Formats/PHPIni” does not exist
./libpangocairo-1.0.so.0: ./libz.so.1: version `ZLIB_1.2.9' not found (required by /nix/store/3ah3k65zwlfz1gjf0fxsv0ygz90b7g8i-libpng-apng-1.6.29/lib/libpng16.so.16)
./libpangoft2-1.0.so: ./libz.so.1: version `ZLIB_1.2.9' not found (required by /nix/store/3ah3k65zwlfz1gjf0fxsv0ygz90b7g8i-libpng-apng-1.6.29/lib/libpng16.so.16)
./libpangocairo-1.0.so: ./libz.so.1: version `ZLIB_1.2.9' not found (required by /nix/store/3ah3k65zwlfz1gjf0fxsv0ygz90b7g8i-libpng-apng-1.6.29/lib/libpng16.so.16)
./libpangoxft-1.0.so.0: ./libz.so.1: version `ZLIB_1.2.9' not found (required by /nix/store/3ah3k65zwlfz1gjf0fxsv0ygz90b7g8i-libpng-apng-1.6.29/lib/libpng16.so.16)
./libpangoxft-1.0.so: ./libz.so.1: version `ZLIB_1.2.9' not found (required by /nix/store/3ah3k65zwlfz1gjf0fxsv0ygz90b7g8i-libpng-apng-1.6.29/lib/libpng16.so.16)
./libpangoft2-1.0.so.0: ./libz.so.1: version `ZLIB_1.2.9' not found (required by /nix/store/3ah3k65zwlfz1gjf0fxsv0ygz90b7g8i-libpng-apng-1.6.29/lib/libpng16.so.16)
```
The above library files are located in `$out/libexec/Mathematica/SystemFiles/Libraries/Linux-x86-64/`, which also has`libz.so.1` shipped with Mathematica.

I have confirmed that just deleting this `libz.so.1` works fine. Instead of lines 98-102:
```sh
    # Delete shipped libz
    case `uname -m` in
      i?86)
        zlib_dirs="Linux" ;;
      x86_64)
        zlib_dirs="Linux-x86-64 Linux" ;;
      armv?l)
        zlib_dirs="Linux-ARM" ;;
    esac
    if [ -n "''${zlib_dirs}" ]; then
      for dir in ''${zlib_dirs}; do
        rm $out/libexec/Mathematica/SystemFiles/Libraries/''${dir}/libz.so.1
      done
    fi

```